core/domain/email_manager_test.py

```diff
@@ -2645,3 +2646,365 @@ def test_correct_moderator_action_email_sent(self):
                 email_intent, exploration_title, email_html_body)
         messages = self.mail_stub.get_sent_messages(to=self.RECIPIENT_EMAIL)
         self.assertEqual(len(messages), 1)
+
+
+class CommunityReviewerEmailTest(test_utils.GenericTestBase):
+    """Test for assignment and removal of reviewer in community."""
+    TRANSLATION_REVIEWER_EMAIL = 'translationreviewer@example.com'
+    VOICEOVER_REVIEWER_EMAIL = 'voiceoverreviewer@example.com'
+    QUESTION_REVIEWER_EMAIL = 'questionreviewer@example.com'
+
+    def setUp(self):
+        super(CommunityReviewerEmailTest, self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.signup(self.TRANSLATION_REVIEWER_EMAIL, 'translator')
+        self.signup(self.VOICEOVER_REVIEWER_EMAIL, 'voiceartist')
+        self.signup(self.QUESTION_REVIEWER_EMAIL, 'question')
+
+        self.translation_reviewer_id = self.get_user_id_from_email(
+            self.TRANSLATION_REVIEWER_EMAIL)
+        user_services.update_email_preferences(
+            self.translation_reviewer_id, True, False, False, False)
+        self.voiceover_reviewer_id = self.get_user_id_from_email(
+            self.VOICEOVER_REVIEWER_EMAIL)
+        user_services.update_email_preferences(
+            self.voiceover_reviewer_id, True, False, False, False)
+        self.question_reviewer_id = self.get_user_id_from_email(
+            self.QUESTION_REVIEWER_EMAIL)
+        user_services.update_email_preferences(
+            self.question_reviewer_id, True, False, False, False)
+
+        self.can_send_emails_ctx = self.swap(
+            feconf, 'CAN_SEND_EMAILS', True)
+        self.can_not_send_emails_ctx = self.swap(
+            feconf, 'CAN_SEND_EMAILS', False)
+
+    def test_assign_translation_reviewer_email_for_can_send_emails_is_false(
+            self):
+        with self.can_not_send_emails_ctx:
+            email_manager.send_email_to_new_community_reviewer(
+                self.translation_reviewer_id,
+                constants.REVIEW_CATEGORY_TRANSLATION, language_code='hi')
+
+        messages = self.mail_stub.get_sent_messages(
+            to=self.TRANSLATION_REVIEWER_EMAIL)
+        self.assertEqual(len(messages), 0)
+
+    def test_assign_translation_reviewer_email_for_invalid_review_category(
+            self):
+        with self.assertRaisesRegexp(Exception, 'Invalid review_category'):
+            email_manager.send_email_to_new_community_reviewer(
+                self.translation_reviewer_id, 'invalid_category')
+
+    def test_all_category_keys_present_in_new_reviewer_email_data_constant(
```

Comment:
For both the new constants, you should loop over the keys and validate the schema of the value.

Reply:
Done! (I've all the schema test for once constant in one test should I create a separate test for each value in the constant?)